src/components/drop/DomPositionCalculator.ts

```diff
@@ -0,0 +1,162 @@
+import {CSSProperties} from 'react';

```

Comment:
add `dropUIPosition: IDropUIPosition` for each position to calculate the next position if the button position do not have space to render drop inside the bounding limit